extensions/azurePublish/src/luisAndQnA.ts

```diff
@@ -6,190 +6,76 @@ import { promisify } from 'util';
 import { BotProjectDeployLoggerType } from './botProjectLoggerType';
-import { luImportResolverGenerator } from '@bfc/shared/lib/luBuildResolver';
 
-const crossTrainer = require('@microsoft/bf-lu/lib/parser/cross-train/crossTrainer.js');
-const luBuild = require('@microsoft/bf-lu/lib/parser/lubuild/builder.js');
-const qnaBuild = require('@microsoft/bf-lu/lib/parser/qnabuild/builder.js');
 const readdir: any = promisify(fs.readdir);
 
-export interface PublishConfig {
-  // Logger
-  logger: (string) => any;
-  projPath: string;
-  [key: string]: any;
-}
-
-const INTERRUPTION = 'interruption';
-
-export class LuisAndQnaPublish {
-  private logger: (string) => any;
-  private remoteBotPath: string;
-  private generatedFolder: string;
-  private interruptionFolderPath: string;
-  private crossTrainConfig: ICrossTrainConfig;
-
-  constructor(config: PublishConfig) {
-    this.logger = config.logger;
-    // path to the ready to deploy generated folder
-    this.remoteBotPath = path.join(config.projPath, 'ComposerDialogs');
-    this.generatedFolder = path.join(this.remoteBotPath, 'generated');
-    this.interruptionFolderPath = path.join(this.generatedFolder, INTERRUPTION);
-
-    // Cross Train config
-    this.crossTrainConfig = {
-      rootIds: [],
-      triggerRules: {},
-      intentName: '_Interruption',
-      verbose: true,
-      botName: '',
-    };
-  }
+const botPath = (projPath: string) => path.join(projPath, 'ComposerDialogs')
 
-  /*******************************************************************************************************************************/
-  /* This section has to do with publishing LU files to LUIS
-  /*******************************************************************************************************************************/
+type QnaConfigType = {
+  subscriptionKey: string;
+  qnaRegion: string | 'westus';
+};
 
-  /**
-   * return an array of all the files in a given directory
-   * @param dir
-   */
-  private async getFiles(dir: string): Promise<string[]> {
-    const dirents = await readdir(dir, { withFileTypes: true });
-    const files = await Promise.all(
-      dirents.map((dirent) => {
-        const res = path.resolve(dir, dirent.name);
-        return dirent.isDirectory() ? this.getFiles(res) : res;
-      })
-    );
-    return Array.prototype.concat(...files);
-  }
-
-  /**
-   * Helper function to get the appropriate account out of a list of accounts
-   * @param accounts
-   * @param filter
-   */
-  private getAccount(accounts: any, filter: string) {
-    for (const account of accounts) {
-      if (account.AccountName === filter) {
-        return account;
-      }
-    }
-  }
+type Resources = {
+  luResources: string[];
+  qnaResources: string[];
+}
 
-  private notEmptyModel(file: string) {
-    return fs.readFileSync(file).length > 0;
-  }
+type BuildSettingType = {
+  luis: ILuisConfig,
+  qna: QnaConfigType
+} & Resources
```

Comment:
>& Resources [](start = 2, length = 11)

minor: for perf reasons listed [here:](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections), is it easy to switch to interfaces?